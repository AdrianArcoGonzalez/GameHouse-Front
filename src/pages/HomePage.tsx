import { ToastContainer } from "react-toastify";
import heroTexts from "../components/HeroSection/heroData/heroData";
import HeroSection from "../components/HeroSection/HeroSection";
import "react-toastify/dist/ReactToastify.css";
import Games from "../components/Games/Games";
import HomePageStyled from "./HomePageStyled";
import JoinUs from "../components/JoinUs/JoinUs";
import GameDetails from "../components/GameDetails/GameDetails";
import { mockGame } from "../mocks/mockGame";

const HomePage = (): JSX.Element => {
  return (
    <>
      <HomePageStyled>
        <ToastContainer />
        <HeroSection text={heroTexts.home} srcImage={"images/home.webp"} />
        <h2 className="games-list__title">Comunity Games</h2>
        <Games />
        <JoinUs />
      </HomePageStyled>
      <GameDetails game={mockGame} />
    </>
  );
};

export default HomePage;
