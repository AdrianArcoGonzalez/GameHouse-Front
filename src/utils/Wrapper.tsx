import * as reactRedux from "react-redux";
import { store } from "../store/store";

export interface WrapperProps {
  children: JSX.Element | JSX.Element[];
}

export const Wrapper = ({ children }: WrapperProps): JSX.Element => {
  return <reactRedux.Provider store={store}>{children}</reactRedux.Provider>;
};
