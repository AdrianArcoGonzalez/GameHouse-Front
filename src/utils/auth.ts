import jwt from "jwt-decode";

const decodeToken = (token: string) => {
  const payloadToken: {
    id: string;
    username: string;
    iat: number;
  } = jwt(token);

  const user = {
    token: token,
    id: payloadToken.id,
    username: payloadToken.username,
    isLogged: true,
  };
  return user;
};

export default decodeToken;
