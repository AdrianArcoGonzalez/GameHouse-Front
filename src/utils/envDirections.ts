export const backUrl = process.env.REACT_APP_URL_BACK

export const urlsBack = {
  default: `${backUrl}games/games/`,
  getByOwner: `${backUrl}games/games/my-collection/`,
  getByCategory: `${backUrl}games/games/category/`,

}
