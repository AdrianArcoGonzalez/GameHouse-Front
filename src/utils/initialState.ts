import decodeToken from "./auth";

const emptyUser = {
  id: "",
  username: "",
  token: "",
  isLogged: false,
};
const initialState = () => {
  const token = localStorage.getItem("token") as string;

  if (token) {
    return decodeToken(token);
  }

  return emptyUser;
};

export default initialState;
