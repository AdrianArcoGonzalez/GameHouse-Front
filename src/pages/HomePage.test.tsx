import { Wrapper } from "../utils/Wrapper";
import TestRenderer from "react-test-renderer";
import HomePage from "./HomePage";

describe("Given a Home Page component", () => {
  describe("When it's instantiated", () => {
    test("Then it should match the html from Home page", () => {
      const expectedHomePage = TestRenderer.create(
        <Wrapper>
          <HomePage />
        </Wrapper>
      );

      expect(expectedHomePage).toMatchSnapshot();
    });
  });
});
