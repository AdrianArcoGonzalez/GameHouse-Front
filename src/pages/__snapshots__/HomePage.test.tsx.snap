// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Home Page component When it's instantiated Then it should match the html from Home page 1`] = `
<div
  className="sc-hKMtZM iMoGjk"
>
  <div
    className="Toastify"
  />
  <section
    className="sc-bczRLJ dVNtAm"
    data-testid="section"
  >
    <img
      alt="backgroundImage"
      className="hero__image"
      height="200px"
      src="images/home.webp"
      width="100vw"
    />
    <div
      className="hero-container"
    >
      <p
        className="hero-container__sentence"
      >
        Cuanto más te acerques a la luz, mayor será tu sombra (Kingdom Hearts)
      </p>
    </div>
  </section>
  <h2
    className="games-list__title"
  >
    Comunity Games
  </h2>
  <section
    className="sc-dkzDqf iZklhZ"
  >
    <ul
      className="games-list"
    />
  </section>
  <section
    className="sc-eCYdqJ dFhFLD"
  >
    <img
      alt="Video Game characters riding birds running on a field"
      className="join-us__image"
      height="300px"
      src="/images/joinComunity.jpg"
      width="320px"
    />
    <div
      className="join-us__info-container"
    >
      <h3
        className="join-us__title"
      >
        Join the community!
      </h3>
      <p
        className="join-us__text"
      >
        Become a soldier of the GameHouse Army. Make your own reviews and give and receive feedback from other soldiers!
      </p>
      <button
        className="join-us__button"
      >
        <a
          className="join-us__navigate"
          href="/register"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </button>
      <button
        className="join-us__button"
      >
        <a
          className="join-us__navigate"
          href="/login"
          onClick={[Function]}
        >
          Sign In
        </a>
      </button>
    </div>
  </section>
</div>
`;
